apis/v1alpha1/gateway_types.go

```diff
@@ -466,54 +468,45 @@ type RouteNamespaces struct {
 
 // GatewayAddress describes an address that can be bound to a Gateway.
 type GatewayAddress struct {
-	// Type of the Address. This is either "IPAddress" or "NamedAddress".
+	// Type of the address.
 	//
 	// Support: Extended
 	//
 	// +kubebuilder:default=IPAddress
 	Type AddressType `json:"type,omitempty"`
 
-	// Value. Examples: "1.2.3.4", "128::1", "my-ip-address". Validity of the
-	// values will depend on `Type` and support by the controller.
+	// Value of the address.
+	// The validity of the  values will depend on the type and support by the controller.
+	//
+	// Examples: "1.2.3.4", "128::1", "my-ip-address".
 	//
 	// +kubebuilder:validation:MinLength=1
 	// +kubebuilder:validation:MaxLength=253
 	Value string `json:"value"`
 }
 
 // AddressType defines how a network address is represented as a text string.
-// Valid AddressType values are:
 //
-// * "IPAddress": an IPv4 or IPv6 address
-// * "NamedAddress": an opaque identifier. The interpretation
-// of the name is dependent on the controller. If a NamedAddress is
-// requested but unsupported by an implementation, the controller
-// should raise the "Detached" listener status condition on the Gateway
-// with the "UnsupportedAddress" reason.
-//
-// A NamedAddress might be a cloud-dependent ID for a "static" or "elastic" IP
+// If the requested address is unsupported, the controller
+// should raise the "Detached" listener status condition on
+// the Gateway with the "UnsupportedAddress" reason.
 //
 // +kubebuilder:validation:Enum=IPAddress;NamedAddress
 type AddressType string
 
 const (
-	// IPAddressType a textual representation of a numeric IP
+	// A textual representation of a numeric IP
 	// address. IPv4 addresses must be in dotted-decimal
 	// form. IPv6 addresses must be in a standard IPv6 text
-	// representation (see RFC 5952).
-	//
-	// If the requested address is unsupported, the controller
-	// should raise the "Detached" listener status condition on
-	// the Gateway with the "UnsupportedAddress" reason.
+	// representation (see [RFC 5952](https://tools.ietf.org/html/rfc5952)).
 	//
 	// Support: Extended.
 	IPAddressType AddressType = "IPAddress"
 
-	// NamedAddressType is an opaque identifier. The interpretation
-	// of the name is dependent on the controller. If a NamedAddress is
-	// requested but unsupported by an implementation, the controller
-	// should raise the "Detached" listener status condition on the Gateway
-	// with the "UnsupportedAddress" reason.
+	// An opaque identifier that represents a specific IP address. The interpretation
+	// of the name is dependent on the controller. For example,
+	// a "NamedAddress" might be a cloud-dependent ID for a "static"
+	// or "elastic" IP.
```

Comment:
I think it is best to leave the idiomatic Godoc style "<Identifier> is ..." in here for the sake of consistency across the entire project.

Reply:
The problem with the Go idiom here is that the API isn't the constant t name but the constant value (ie. the value is what you need in the YAML). So using the constant name here is actually misleading readers of the documentation. It's also confusing because the constant names done appear anywhere in the HTML rendering.

Reply:
This is an interesting one. I took a look at the Kubernetes reference docs and they lack consistency here but the godoc style Harry is suggesting above does seem to be more common ([ConfigMap](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#configmap-v1-core) is an example of that). Your proposed change does certainly read better in the reference docs, we just need to determine how heavily we want to prioritize that over other formats.

Reply:
Here's an example of the current docs:

<img width="700" alt="Screen Shot 2021-01-17 at 11 40 11 am" src="https://user-images.githubusercontent.com/9917/104828011-ef7b6400-58b8-11eb-9501-66de071193d7.png">

Looking at "NamedAddress", which is a designated enum value that API clients have to provide, it is described in terms of a "NamedAddressType", a term which is never defined in the documentation and is utterly irrelevant to anyone who doesn't have the Go code in front of them.

In terms of the API, the fact that this constant is available in Go doesn't really matter. What API consumers need to know is the *value* of the constant.